src/backend/dx11/src/lib.rs

```diff
@@ -2037,49 +2156,84 @@ impl command::CommandBuffer<Backend> for CommandBuffer {
                 }
             }
 
-            // TODO: offsets
+            let (vs_offsets, fs_offsets) = generate_graphics_dynamic_constant_buffer_offsets(
+                &*set.layout.bindings,
+                &mut offset_iter,
+                self.context1.is_some()
+            );
+
+            // TODO: This should be the actual buffer length for RBA purposes,
+            //       but that information isn't easily accessible here.
+            let vs_counts = vec![4096_u32; vs_offsets.len()];
```

Comment:
can we have less dynamic allocations in hot functions like this one?
I.e. `vs_offsets`, `vs_counts`, `fs_offsets`, `fs_counts` - all of them could be in some `struct Temp` within the command buffer, just re-used.